### ISSUE TITLE
Liking and boosting posts does not work

### ISSUE THREAD

**Issue by makeworld-the-better-one** [Plume-org/Plume#367]
When I like or boost a post (on the same instance), it just takes me to a `domain.com/posturl/{like,boost}` url, which is just a blank page. When I reload the blog's page, the like or boost count still remains at 0, and it seems I haven't affected anything.

**Comment by fdb-hiroshima** [comment]
On what instance is that, and what browser are you using? I tried on fediverse.blog with Firefox and it worked just fine.

**Comment by marek-lach** [comment]
I have now tested boosting, and liking and following a user from the fediverse.blog instance, and I found that I can boost and like their post, but cannot follow them at the moment.

Their account is hosted on the cafe.sunbeam.city instance, therefore I think that the cafe.sunbeam.city server is just misconfigured somehow, or their installation of Plume was not done correctly.

I cannot like, or boost posts when logged into the cafe.sunbeam.city instance. I am not sure if they use the latest master branch, or not. But updating fediverse.blog from the latest repo may give some idea if the bug is in the master branch, or if it's just a specific server misconfiguration.

<img width="1166" alt="screenshot 2018-12-21 at 14 02 56" src="https://user-images.githubusercontent.com/45913977/50344312-cb5ac680-052a-11e9-95b3-0863909ef236.png">

<img width="1280" alt="screenshot 2018-12-21 at 14 04 31" src="https://user-images.githubusercontent.com/45913977/50344327-db72a600-052a-11e9-86ad-a5e2eda7be9c.png">

Simply put: The inability to boost/like/follow users IS NOT present in the branch that is currently in use at the fediverse.blog instance, whichever branch is currently in use there.

**Comment by f0x52** [comment]
@marek-lach cafe.sunbeam.city is running the latest master
The console regularly gives errors, which might be related:
```rust
Dec 21 14:39:49 ns3059072 plume[21861]: thread '<unnamed>' panicked at 'instance::shared_inbox: user error', src/libcore/option.rs:1008:5
Dec 21 14:40:06 ns3059072 plume[21861]: POST /inbox/ application/activity+json:
Dec 21 14:40:06 ns3059072 plume[21861]:     => Matched: POST /inbox (shared_inbox)
Dec 21 14:40:06 ns3059072 plume[21861]: thread '<unnamed>' panicked at 'instance::shared_inbox: user error', src/libcore/option.rs:1008:5
Dec 21 14:41:25 ns3059072 plume[21861]: POST /inbox/ application/activity+json:
Dec 21 14:41:25 ns3059072 plume[21861]:     => Matched: POST /inbox (shared_inbox)
Dec 21 14:41:26 ns3059072 plume[21861]: thread '<unnamed>' panicked at 'instance::shared_inbox: user error', src/libcore/option.rs:1008:5
Dec 21 14:41:46 ns3059072 plume[21861]: POST /inbox/ application/activity+json:
Dec 21 14:41:46 ns3059072 plume[21861]:     => Matched: POST /inbox (shared_inbox)
Dec 21 14:41:46 ns3059072 plume[21861]: thread '<unnamed>' panicked at 'instance::shared_inbox: user error', src/libcore/option.rs:1008:5
```

**Comment by marek-lach** [comment]
@fdb-hiroshima So, upon further investigation I found that:

1.) It is possible to boost and like a post from fediverse.blog and other instances like plume.mastodon.host, which is not possible on cafe.sunbeam.city, which, as @f0x52 says is running the latest master.

2.) While it is possible to boost and like a post from instances like fediverse.blog, or plume.mastodon.host, it isn't possible to follow people even from these instances, because upon trying to follow a user on fediverse.blog, or plume.mastodon.host, the following error appears:

<img width="1166" alt="screenshot 2018-12-21 at 14 02 56" src="https://user-images.githubusercontent.com/45913977/50351849-a9217280-0543-11e9-87e4-57e56fbea423.png">
<img width="1280" alt="screenshot 2018-12-21 at 17 01 38" src="https://user-images.githubusercontent.com/45913977/50351861-b76f8e80-0543-11e9-98bc-909c2f44cce5.png">
<img width="1280" alt="screenshot 2018-12-21 at 16 57 52" src="https://user-images.githubusercontent.com/45913977/50351880-c5bdaa80-0543-11e9-9b14-59518524baf8.png">

The latest master branch, running on cafe.sunbeam.city displays this instead when trying to follow someone:

<img width="1280" alt="screenshot 2018-12-21 at 14 04 31" src="https://user-images.githubusercontent.com/45913977/50351942-fbfb2a00-0543-11e9-84ee-56ac43eefad6.png">

I hope this helps with diagnosing the problem better.

**Comment by fdb-hiroshima** [comment]
This is odd, when trying to follow the very same account as your first screenshot (RadioAngel) from my own account on fediverse.blog, I'm not getting an error. How are testing? Are you using the follow link or entering the url manually?

**Comment by marek-lach** [comment]
@fdb-hiroshima This is indeed very odd. But I tested in a different browser now (Opera, previously Firefox) and got the same result as before. Even trying to follow an entirely different user returned the same 'page not found error':

<img width="1280" alt="screenshot 2018-12-22 at 00 33 31" src="https://user-images.githubusercontent.com/45913977/50367696-413e4c80-0582-11e9-876a-108f36f0a3b5.png">

I am thinking it may have something to do with access rights? Like, there must be a difference between various administrative rights that an administrator has, versus a typical user has?
Something like that... I have no idea what else it could be...

**Comment by makeworld-the-better-one** [comment]
@marek-lach This seems to be a different issue that the original one I proposed. Do you or @fdb-hiroshima have any ideas, based on cafe.sunbeam.city's version and error log that @f0x52 provided? Would a reinstall work? How was fediverse.blog installed?

**Comment by marek-lach** [comment]
@makeworld-the-better-one I don't know myself, as I cannot code myself, as @fdb-hiroshima can, but I think these two problems are closely related, so solving one may well solve the other?

**Comment by fdb-hiroshima** [comment]
@marek-lach I don't have an admin account, and I'd be surprised it was something like that anyway, we check for privileges only on very few routes. Under Firefox, if you go in the developers tools (press F12 on while on the page), and switch to the Network tab before loading the following page, when opening the follow link, what Method the query to the follow file did your browser use?
@makeworld-the-better-one you're right, we should open an other issue to continue the discussion further. The logs @f0x52 shared are symptomatic of a federation problem, which is not the error you observed. What you saw was an error 500 while liking/resharing. The error, in logs, should look something like that:
```
POST /~/someBlog/some-title/like application/x-www-form-urlencoded:
    => Matched: POST /~/<blog>/<slug>/like (create)
thread '<unnamed>' panicked at '[some error message]', src/libcore/option.rs:1008:5
```

**Comment by fdb-hiroshima** [comment]
@makeworld-the-better-one both errors are most likely unrelated, one is an internal server error, an error we did not handle, your's look more like a dead-link, maybe a browser extension rewriting the query or idk what

**Comment by makeworld-the-better-one** [comment]
@fdb-hiroshima So what do you think of reinstalling? Could that fix the issue, since fediverse.blog doesn't have it?

**Comment by marek-lach** [comment]
@fdb-hiroshima From what I can tell, there's a 404, missing document error when I try to follow a user on fediverse.blog.

Before following:

<img width="1280" alt="screenshot 2018-12-22 at 01 06 13" src="https://user-images.githubusercontent.com/45913977/50368189-32599900-0586-11e9-99b1-0fe83dca3ea2.png">

After following:

<img width="1280" alt="screenshot 2018-12-22 at 01 06 31" src="https://user-images.githubusercontent.com/45913977/50368201-4ef5d100-0586-11e9-8c99-e795e0b45a79.png">


This is the error report from Firefox when trying to like a post on cafe.sunbeam.city:

<img width="1280" alt="screenshot 2018-12-22 at 01 11 40" src="https://user-images.githubusercontent.com/45913977/50368232-a8f69680-0586-11e9-81c5-cc24d560d329.png">

I see the 500 internal server error code as the main cause of 'like', or 'boost' not working on cafe.sunbean.city?

Since ''the 500 Internal Server Error is a server-side error, meaning the problem probably isn't with your computer or internet connection but instead with the website's server'', I suspect that reinstalling, or somehow reconfiguring the installation on the servers of cafe.sunbean.city might help?

Although I don't know what would help with the follow problem I have on fediverse.blog :-)

**Comment by fdb-hiroshima** [comment]
@makeworld-the-better-one I think Plume does not have much state else than the db content, reinstalling while keeping the database would probably change nothing, and flushing the db is not an option. I really need the part of the log that correspond to your problem to help in anything.

@marek-lach Are you using something like a browser extension that might rewrite the query or page content? Your browser is emitting a GET request to a resource existing only with the POST verb, and after verification, there is no link going to that resource via GET.

**Comment by marek-lach** [comment]
@fdb-hiroshima Disabled all plugins, cleared all history, still have the same problem both in Firefox, and in freshly installed Chrome.

Here is what Chrome's console gives when trying to like a post on cafe.sunbeam.city, allegedly running the latest 'master' branch: 

<img width="1280" alt="screenshot 2018-12-22 at 16 09 13" src="https://user-images.githubusercontent.com/45913977/50375862-cff3ad80-0604-11e9-8efb-94586516fabd.png">

And this is what Chrome's console gives when liking a post on fediverse.blog, whichever branch that's running:

<img width="1280" alt="screenshot 2018-12-22 at 16 11 33" src="https://user-images.githubusercontent.com/45913977/50375876-08938700-0605-11e9-8288-b89abe2ae046.png">

So where cafe.sunbean.city gives a 404 not found error on like, fediverse.blog succesfully likes a post, with a 303 code on like.

https://en.wikipedia.org/wiki/HTTP_303

**Comment by BaptisteGelez** [comment]
I guess this is an issue with the `ap_url` of the `Like` we try to create (https://github.com/Plume-org/Plume/blob/master/src/routes/likes.rs#L24), because of the constraints that were recently introduced (Fediverse.blog is not yet using this version of the code). This field is probably duplicated for some reason. But I can't be sure this is the issue until we have more precise logs.

**Comment by fdb-hiroshima** [comment]
ok, this is what @BaptisteGelez guessed, which is definitely a bug
